src/Core/FileMonitor/SingleFileMonitorFactory.cs

```diff
@@ -0,0 +1,51 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.ComponentModel.Composition;
+
+namespace SonarLint.VisualStudio.Core.FileMonitor;
+
+public interface ISingleFileMonitorFactory
+{
+    ISingleFileMonitor Create(string filePathToMonitor);
+}
+
+[Export(typeof(ISingleFileMonitorFactory))]
+[PartCreationPolicy(CreationPolicy.Shared)]
+internal class SingleFileMonitorFactory : ISingleFileMonitorFactory
+{
+    private readonly ILogger logger;
+
+    [ImportingConstructor]
+    public SingleFileMonitorFactory(ILogger logger)
+    {
+        this.logger = logger;
+    }
+
+    public ISingleFileMonitor Create(string filePathToMonitor)
+    {
+        if (filePathToMonitor == null)
+        {
+            throw new ArgumentNullException(nameof(filePathToMonitor));
+        }
+
+        return new SingleFileMonitor(filePathToMonitor, logger);
```

Comment:
Might move FileSystem and FileSystemWatcherFactory creation to here?

Reply:
Does it bring any value? I would need to touch/refactor a lot of tests, and I am not sure the change has any value that would be worth investing the effort into it.

Reply:
The value would be a clear separation of concerns. The factory should manage all of the dependencies for the class. 

You can create a ticket in the backlog for this

Reply:
Did as suggested